### PR TITLE
add nui build-sass

### PR DESCRIPTION
**Pull request by doramatadora**
Just an idea: could we have a `nui build-sass` command? It would be great to have a way to build css only locally. Doing it concurrently with webpack hangs for a lot of us.

Or a `--sass-only` flag on `nui build`?

### CODE REVIEW

**Comment by leggsimon** [comment]
Are you proposing the same for JS as well?

Just in case you weren't aware, `nui watch` will rebuild sass only on save if you are editing that.

**Comment by alexnaish** [comment]
The issue for me isn't whats built during the `watch` task, its the problem that 9/10 times I just need to build `head.css` in order to run my unit-tests for example and both `make build` and `make watch` often fail to build SASS first time.

**Comment by leggsimon** [comment]
I think I would prefer a `--sass-only`/`--js-only` option to be honest @doramatadora

File: build/app/cli.js [hunk 1]
```diff
@@ -54,16 +54,29 @@ program
...
+	.option('--js-only', 'Builds JavaScript only')
 	.action(options => {
 
 		devAdvice();
 		const buildStartTime = Date.now();
-		const cssBuildCommands = cssEntryPoints.map(([target, entry]) => {
-			const script = './node_modules/@financial-times/n-ui/scripts/build-sass.sh';
-			return `'${script} ${entry} ${target}'`;
-		}).join(' ');
+		let concurrentComands = [];
```
Comment (alexnaish) [inline build/app/cli.js]: Picky (sorry!) but commands has 2 `m`s

File: build/app/cli.js [hunk 2]
```diff
@@ -54,49 +54,29 @@ program
...
-				const buildTime = Date.now() - buildStartTime;
+		const script = './node_modules/@financial-times/n-ui/scripts/build-sass.sh';
+		const commands = {
+			'js-only': `'webpack --bail --config ${webpackConfPath} ${options.production ? '-p' : ''}'`,
+			'sass-only': cssEntryPoints.map(([target, entry]) => `'${script} ${entry} ${target}'`).join(' ')
+		};
 
-				// Don't send metrics from CircleCI builds
-				if (!process.env.CIRCLECI) {
-					sendBuildMetrics(appPackageJson.name, buildTime);
```
Comment (leggsimon) [inline build/app/cli.js]: Can we keep this in please for a full build.

File: build/app/cli.js [hunk 3]
```diff
@@ -54,49 +54,29 @@ program
...
-	.option('--production', 'Builds with production settings')
-	.action(options => {
-
-		devAdvice();
-		const buildStartTime = Date.now();
-		const cssBuildCommands = cssEntryPoints.map(([target, entry]) => {
-			const script = './node_modules/@financial-times/n-ui/scripts/build-sass.sh';
-			return `'${script} ${entry} ${target}'`;
-		}).join(' ');
+		concurrentComands = concurrentComands.length ? concurrentComands : Object.values(commands);
```
Comment (leggsimon) [inline build/app/cli.js]: Using `Object.values()` will only work if you're using > Node 7.5.

File: build/app/cli.js [hunk 4]
```diff
@@ -54,16 +54,29 @@ program
...
+			'sass-only': cssEntryPoints.map(([target, entry]) => `'${script} ${entry} ${target}'`).join(' ')
+		};
+
+		for(let key in commands) {
+			if(!!options[key]) {
+				concurrentCommands.push(commands[key]);
+			}
+		}
+
+		concurrentCommands = concurrentCommands.length ? concurrentCommands : Object.values(commands);
```
Comment (leggsimon) [inline build/app/cli.js]: `Object.values()` will only work for users on > Node 7.5 😢

**Pull request merged**